pkg/operator/controller/ingress/status.go

```diff
@@ -30,6 +30,16 @@ import (
 // clock is to enable unit testing
 var clock utilclock.Clock = utilclock.RealClock{}
 
+// expectedCondition contains a condition that is expected to be checked when
+// determining Available or Degraded status of the ingress controller
+type expectedCondition struct {
+	condition        string
+	status           operatorv1.ConditionStatus
+	ifConditionsTrue []string // ifConditionsTrue is a list of dependent conditions that should be be true as well
+	gracePeriod      time.Duration
+	mustHave         bool // mustHave determines whether this expectedCondition must be there
```

Comment:
```suggestion
	// mustHave specifies whether this expectedCondition must be there.
	mustHave         bool
```
Is it true that `computeIngressAvailableCondition` will specify `mustHave: true` for all its expected conditions whereas `computeIngressDegradedCondition` will specify `mustHave: false` for all its expected conditions?  If so, would it make more sense for `checkAnnotatedConditions` to have an `ignoreMissingConditions` parameter?

Reply:
For now that is true, but I could envision a future use where an individual condition of a set might be a "mustHave".  I thought it best to leave it individually specified, but don't mind changing that if needed.

Reply:
This has been removed.